### DISCUSSION TITLE
Lesson 10: Problem to lunch local RPC.

### DISCUSSION THREAD

**Discussion by vicentemunozlh** [smartcontractkit/full-blockchain-solidity-course-py#1692] [Q&A] [answered]
Hello, I am having problems with the ganache when trying to lunch aave_borrow.py . I understand that this has happened to other people before but I don't understand how to change the node to 16 as some replays say. Although I notice that the command in the error says Command: ganache-cli while the command for ganache is ganache not ganache-cli. I am lost, thankfull for any help if possible!

```{python}
~\demos\aave_brownie_py> brownie run scripts/aave_borrow.py --network mainnet-fork-dev
INFO: Could not find files for the given pattern(s).
Brownie v1.19.1 - Python development framework for Ethereum

AaveBrowniePyProject is the active project.

Launching 'ganache-cli.cmd --chain.vmErrorsOnRPCResponse true --wallet.totalAccounts 10 --fork.url https://infura.io/v3/b777276e91ea477e884a10525a242daa --wallet.mnemonic brownie --server.port 8545 --hardfork istanbul'...
  File "~\Local\Programs\Python\Python310\lib\site-packages\brownie\_cli\__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "~\Local\Programs\Python\Python310\lib\site-packages\brownie\_cli\run.py", line 45, in main
    network.connect(CONFIG.argv["network"])
  File "~\Local\Programs\Python\Python310\lib\site-packages\brownie\network\main.py", line 50, in connect
    rpc.launch(active["cmd"], **active["cmd_settings"])
  File "~\Local\Programs\Python\Python310\lib\site-packages\brownie\network\rpc\__init__.py", line 94, in launch
    raise RPCProcessError(cmd, uri)
RPCProcessError: Unable to launch local RPC client.
Command: ganache-cli
URI: http://127.0.0.1:8545

**Comment by cromewar** [comment, marked as answer]
Hello @vicentemunozlh
The reason is the npm package for ganache changed recently from ganache-cli to just ganache, so deleting the network and adding it again on brownie should be enough to solve this, of course remember to set cmd=ganache.
Cheers.

**Reply by vicentemunozlh** [reply]
that worked perfectly, thanks!

**Reply by cromewar** [reply]
I'm glad is solved now, feel free to make any other question.

**Reply by vicentemunozlh** [reply]
Hello again @cromewar, the same error came trough, yesterday deleting and adding the network worked, but now this method isn't working. I tried using the same network before deleting it but it didn't work.  Is there anything else that might be the problem or solution?

```{python}
PS \demos\aave_brownie_py> brownie run scripts/aave_borrow.py --network mainnet-fork-dev                                                                           INFO: Could not find files for the given pattern(s).
Brownie v1.19.1 - Python development framework for Ethereum

AaveBrowniePyProject is the active project.

Launching 'ganache.cmd --chain.vmErrorsOnRPCResponse true --wallet.totalAccounts 10 --fork.url https://eth-mainnet.g.alchemy.com/v2/m1Wa9j4-EWtW1Sj3-UN_xw0kbdTPqrpU --wallet.mnemonic brownie --server.port 8545 --hardfork istanbul'...
  File "l\Programs\Python\Python310\lib\site-packages\brownie\_cli\__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "\Programs\Python\Python310\lib\site-packages\brownie\_cli\run.py", line 45, in main
    network.connect(CONFIG.argv["network"])
  File "\Programs\Python\Python310\lib\site-packages\brownie\network\main.py", line 50, in connect
    rpc.launch(active["cmd"], **active["cmd_settings"])
  File "\Programs\Python\Python310\lib\site-packages\brownie\network\rpc\__init__.py", line 94, in launch
    raise RPCProcessError(cmd, uri)
RPCProcessError: Unable to launch local RPC client.
Command: ganache
URI: http://127.0.0.1:8545
``` 

This is the error showed by  the alchemy app. 

![alchemy error](https://user-images.githubusercontent.com/91915615/188960866-1f885e17-f6b7-4a0e-aa8e-f36e4f8e13b8.png)

**Reply by vicentemunozlh** [reply]
I was able to fix it by adding the json file package.json . I don't now if I deleted it by mistake or what happened, but I only had an automatic json file named package-lok.json with this inside:

```{python}
{
  "name": "aave_brownie_py",
  "lockfileVersion": 2,
  "requires": true,
  "packages": {}
}
```
if someone knows why this happened may be helpful for future mistakes!

**Reply by cromewar** [reply]
Hello @vicentemunozlh 
Well to avoid this on the future, install ganache globally:
`npm i -g ganache`
so you will not have to install it each time for all the project you are working with.

**Reply by vicentemunozlh** [reply]
Perfect. Thank you @cromewar .
